GeoQuiz/app/src/main/java/com/example/geoquiz/QuizActivity.kt

```diff
@@ -52,43 +72,113 @@ class QuizActivity : AppCompatActivity() {
             /*val toast=Toast.makeText(this, "Correct",Toast.LENGTH_SHORT)
             toast.setGravity(Gravity.TOP,0,20)
             toast.show()*/
+
             checkAnswer(true)
+
         }
 
         mFalseButton.setOnClickListener {
+
             checkAnswer(false)
+
         }
 
+
         mNextButton.setOnClickListener {
             mCurrentIndex=(mCurrentIndex+1)%mQuestionBank.size
             /*val question=mQuestionBank[mCurrentIndex].textResId
             mQuestionTextView.setText(question)*/
+
             updateQuestion()
+
+
         }
 
         mPrevButton.setOnClickListener {
             mCurrentIndex=(mCurrentIndex-1)%mQuestionBank.size
+            if(mCurrentIndex<0){
+                mCurrentIndex=mQuestionBank.size-1
+            }
             updateQuestion()
+
         }
 
         updateQuestion()
+
+        scoreButton.setOnClickListener {
+
+            for( i in correctAnswers){
+                score += 1;
+            }
+
+
+            Toast.makeText(this,"Your score is: ${(score.toDouble()/4)*100}%",Toast.LENGTH_SHORT).show()
```

Comment:
Very nice use of Kotlin's string template! Still, I have to ask: why divide by four? Would that still work if someone would add another question to the list? If not, how can this be improved?

Reply:
It's best to avoid doing computations to the values in the string template. Especially ones that involve more than one operation.
Please use a val for this computation with a suggestive name.